tensorflow/python/summary/writer/event_file_writer.py

```diff
@@ -108,7 +112,9 @@ def close(self):
 
     Call this method when you do not need the summary writer anymore.
     """
+    self._worker.stop()
     self.flush()
+    self._worker.join()
```

Comment:
I think the following test will almost always block forever:

```python
e = EventFileWriter(...)
time.sleep(1.0)
e.close()
```

The standard approach here would be to send a sentinel element.

(I think I made a similar comment on #4860, which unfortunately was dropped. It would be good to fix this problem, so I'd encourage you to revise this and add some tests that would catch this misbehavior.)